rskj-core/src/test/java/co/rsk/peg/lockingcap/LockingCapSupportImplTest.java

```diff
@@ -0,0 +1,302 @@
+
+        // Assert
+        assertFalse(actualResult);
+        assertEquals(Optional.of(constants.getInitialValue()), lockingCapSupport.getLockingCap());
+    }
+
+    @Test
+    void increaseLockingCap_prePapyrus200_whenLockingCapIsNotSet_shouldReturnFalse() {
+        // Arrange
+        activations = ActivationConfigsForTest.wasabi100().forBlock(0);
+        lockingCapSupport = new LockingCapSupportImpl(lockingCapStorageProvider, activations, constants, signatureCache);
+        Coin newLockingCap = constants.getInitialValue().add(Coin.SATOSHI);
+        Transaction tx = TransactionUtils.getTransactionFromCaller(signatureCache, LockingCapCaller.AUTHORIZED.getRskAddress());
+
+        // Act
+        boolean actualResult = lockingCapSupport.increaseLockingCap(tx, newLockingCap);
+
+        // Assert
+        assertFalse(actualResult);
+        assertEquals(Optional.empty(), lockingCapSupport.getLockingCap());
+    }
+
+    @Test
+    void save_whenIsIncreasedLockingCapValue_shouldSaveLockingCap() {
+        // Arrange
+        Coin newLockingCap = constants.getInitialValue().add(Coin.SATOSHI);
+        Transaction tx = TransactionUtils.getTransactionFromCaller(signatureCache, LockingCapCaller.AUTHORIZED.getRskAddress());
+        lockingCapSupport.increaseLockingCap(tx, newLockingCap);
+
+        // Act
+        lockingCapSupport.save();
+
+        // Assert
+        // Recreate LockingCapSupport to load the previous value from storage and make sure it was saved
+        lockingCapStorageProvider = new LockingCapStorageProviderImpl(bridgeStorageAccessor);
+        lockingCapSupport = new LockingCapSupportImpl(lockingCapStorageProvider, activations, constants, signatureCache);
+        assertEquals(Optional.of(newLockingCap), lockingCapStorageProvider.getLockingCap(activations));
+    }
+
+    @Test
+    void save_prePapyrus200_whenIsAttemptedIncreaseLockingCapValue_shouldNotSaveLockingCap() {
+        // Arrange
+        activations = ActivationConfigsForTest.wasabi100().forBlock(0);
+        lockingCapSupport = new LockingCapSupportImpl(lockingCapStorageProvider, activations, constants, signatureCache);
+        Coin newLockingCap = constants.getInitialValue().add(Coin.SATOSHI);
+        Transaction tx = TransactionUtils.getTransactionFromCaller(signatureCache, LockingCapCaller.AUTHORIZED.getRskAddress());
+        lockingCapSupport.increaseLockingCap(tx, newLockingCap);
+
+        // Act
+        lockingCapSupport.save();
+
+        // Assert
+        // Recreate LockingCapSupport to load the previous value from storage and make sure it was not saved
+        lockingCapStorageProvider = new LockingCapStorageProviderImpl(bridgeStorageAccessor);
+        lockingCapSupport = new LockingCapSupportImpl(lockingCapStorageProvider, activations, constants, signatureCache);
+        assertEquals(Optional.empty(), lockingCapStorageProvider.getLockingCap(activations));
```

Comment:
not sure about these two tests. The `save` method is just calling the one from the storage provider, so we should just assert that.
Check how this was done [here](https://github.com/rsksmart/rskj/blob/feature/locking-cap-support-test/rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java#L2183)

Reply:
I just made a change: using the `getLockingCap` method from `LockingCapSupport` instead of `LockingCapStorageProvider`.

@julia-zack In these tests, we make sure that the new `LockingCap` value is saved correctly. In the case of `prePapyrus200`, we make sure the opposite. I didn't use `mocks` since we are trying to test closely to real life.